### ISSUE TITLE
Make user can customize  the training process. 

### ISSUE THREAD

**Issue by joyyoj** [sql-machine-learning/elasticdl#759]
Currently, user define keras model and loss/optimizer/feature_columns/... in a file and then submit it. 
However, in the most case, users prefer to define custom training process corresponding their demands. Eg.
```python
model = Model()
model.fit(..)
model.fit(...)
model.evaluate(...)
```
My solution is 
```python
@elastic
class MnistModel(keras.Model):
  pass
```

elastic replaces the fit/evaluate/predict method, like this:
```python
def fit(...): 
       if self.role == "master":
            tasks = self._create_tasks(
                self._training_shards, elasticdl_pb2.TRAINING
            )
            do something such as shuffle tasks
            self.create_training_tasks(self.stage)  # use stage for easy recovery.
            self.wait_for_all_tasks_finished(self.stage)
        else:  # role is worker
            while True:
                task = self.get_train_task(self.stage)
                if no more task:
                    break
                self.handle_task(task)
        self.stage += 1
```

**Comment by chunyang-wen** [comment]
This can be left for a future discussion. It is a little hard for users to define their training process currently. And there seems no use cases.
